pytorch_lightning/utilities/seed.py

```diff
@@ -61,8 +69,26 @@ def seed_everything(seed: Optional[int] = None) -> int:
     np.random.seed(seed)
     torch.manual_seed(seed)
     torch.cuda.manual_seed_all(seed)
+
+    if workers:
+        os.environ["PL_SEED_WORKERS"] = f"{int(workers)}"
+
     return seed
 
 
 def _select_seed_randomly(min_seed_value: int = 0, max_seed_value: int = 255) -> int:
     return random.randint(min_seed_value, max_seed_value)
+
+
+def pl_worker_init_function(worker_id: int) -> None:
+    """
+    The worker_init_fn that Lightning automatically adds to your dataloader if you previously set
+    set the seed with :func:`~pytorch_lightning.utilities.seed.seed_everything`.
+    See also the PyTorch documentation on
+    `randomness in DataLoaders <https://pytorch.org/docs/stable/notes/randomness.html#dataloader>`_.
+    """
+    global_rank = rank_zero_only.rank
```

Comment:
Use `_get_rank` directly?

Reply:
I would say safest for now is to keep as is, because some plugins write to that global var and `_get_rank` only fetches the rank from env vars.

Reply:
Okay. It seems weird to use the `rank_zero_only` fn as the source of truth for rank haha